Test

booungyi opened the pull request.

booungyi committed:
temp

booungyi committed:
temp

booungyi committed:
댓글 APi수정

booungyi committed:
댓글 APi수정

booungyi committed:
댓글 답글 , 수정 , 삭제 . 생성 ,대댓글 생성 , 삭제 , 수정 구현

yulmucha commented on src/service/comments.ts:
```diff
@@ -1,89 +1,115 @@
+  const supabase = createClient();
+  const session = await getSessionOrThrow(supabase);
+  const token = session.access_token;
+  const res = await axios.post(`${API_BASE}/comments`, request, {
```
어라 기존에 axios 안 쓰지 않았나요? fetch()만 써도 될 텐데요

booungyi replied:
axios 를 사용하면 인터셉터를 이용해서 중복되는 url이나 헤더토큰을 를 따로 코드마다 써주지않아도 되서 더 간결해보엿습니당

yulmucha commented on src/types/comments.ts:
```diff
@@ -0,0 +1,35 @@
+  likedByMe: boolean;
+  parentId: number | null; // 부모 댓글 ID (null이면 최상위 댓글)
+  createdAt: string;
+  children: CommentResponse[]; // 자식 댓글 배열
```
👍 백엔드랑 똑같이 맞춘 건 잘했어요.
근데 프론트, 백 둘 다 재귀적이던데, 최대 depth 1(대댓글)까지만 허용하는데 괜히 복잡하게 만드는 거 아닐지 한번 고민해 볼만은 한 것 같네요.

booungyi replied:
백엔드에서 이미 대댓글까지만 허용하긴 하는데 프론트에서 parentId 를 명시해주지 않으면 답글의 답글부분의 버튼을 숨길지 말지를 간단하게 구현할수있는게 빙둘러서 구현해야되서 그렇게했습니다.

yulmucha replied:
CommentResponse 타입 안에서 또 동일한 타입인 CommentResponse가 사용되고 있어서 얘기한 거였습니다.

yulmucha commented on src/components/posts/Post.tsx:
```diff
@@ -144,6 +144,18 @@ export default function Post({ initialPost, postId }: Props) {
+      {/* 댓글 섹션 */}
+      <section className="mt-8 pt-6 border-t border-gray-700">
+        <h2 className="text-xl font-semibold mb-4">댓글</h2>
+        <CommentList
+          userId={initialPost.userId}
+          postId={Number(postId)}
+          sortType="TIME" // 최신순으로 정렬
+          size={10} // 한 페이지에 표시할 댓글 수
+          pageNumber={1} // 시작 페이지
+        />
```
페이지 컴포넌트(서버 컴포넌트)에서 데이터를 확보하여 컴포넌트에게 전달하면,
데이터를 전달받은 컴포넌트에서는 그걸 보여주는 데만 집중.
이런 구조가 관심사가 분리되고 재사용성이 좋아져서 좀 더 깔끔하긴 할 텐데 말이죠.

그런데 그 전에,
src/app/posts/[postId]/page.tsx 페이지 컴포넌트에서 Post 컴포넌트만 사용하고,
Post 컴포넌트에서 CommentList 컴포넌트를 사용하는 구조인데,
페이지 컴포넌트에서 Post 컴포넌트를 사용하여 게시글을 보여주면서 그 아래에 나란히 CommentList 컴포넌트를 사용하여 댓글 리스트를 보여주는 방식이 한 눈에 구조를 파악하며 복잡성을 줄일 수 있어서 더 좋을 것 같아요.

booungyi replied:
CommentList를 posts/[postsId]/page.tsx 로 리펙토링-> 완료
위에서 말하는거는  페이지 진입 시 최초 fetch 를 말하시는건지 이해가잘 안가네요

yulmucha replied:
👍

yulmucha commented on src/components/comments/CommentForm.tsx:
```diff
@@ -1,48 +1,95 @@
+        ...(parentId && { parentId }), // parentId가 있을 경우에만 포함
+      };
+
+      console.log('댓글 요청 데이터:', request);
```
참고로, console.log()는 로그 레벨을 설정할 수 없어서 개발용으로 남기더라도 프로덕션 환경에서도 그대로 출력됩니다.
프론트에도 로깅 라이브러리가 있습니다. 나중에 여유 되면 winston 같은 거 한번 찾아보세요.

booungyi replied:
오류 로그보려고 임시로 작성했던 코드입니다.

booungyi replied:
넵

yulmucha commented on src/components/comments/CommentItem.tsx:
```diff
@@ -1,134 +1,245 @@
+  const [isLiking, setIsLiking] = useState(false);
+  const [error, setError] = useState<string | null>(null);
 
   const handleDelete = async () => {
```
`CommentList`가 `CommentItem`에게 삭제 시 사용할 함수를 전달했음에도
`CommentItem`에서는 그대로 사용하지 않고 코드를 더해 또 함수를 만들어 사용하고 있네요.
`CommentItem`이 전달받은 그대로 사용할 만한 함수를 `CommentList`에서 만들어서 전달할 수는 없을까요?
(데이터 흐름을 예측하기 어렵게 만들며, 디버깅과 유지보수를 복잡하게 해요)

booungyi replied:
CommentList 에서는 API 호출과 상태관리 함수 생성후 전달 -> CommentITem 은 함수를 받고 UI 표시와 상호 작용에만 집중

booungyi committed:
axios 인터셉터 사용

booungyi committed:
CommentList 에서는 API 호출과 상태관리 함수 생성루 전달
CommentItem 은 함수를 받고 UI 표시와 상호작용에만 집중
CommentList를 posts/[postsId]/page.tsx 로 리펙토링

booungyi committed:
axios 인터셉터 사용
CommentList 에서는 API 호출과 상태관리 함수 생성루 전달
CommentItem 은 함수를 받고 UI 표시와 상호작용에만 집중
CommentList를 posts/[postsId]/page.tsx 로 리펙토링

booungyi committed:
Merge remote-tracking branch 'origin/test' into test

yulmucha commented on src/service/apiClinet.ts:
```diff
@@ -0,0 +1,18 @@
+import { createClient } from '@/utils/supabase/client';
+import { getSessionOrThrow } from '@/utils/auth';
+
+const apiClient = axios.create({
+  baseURL: process.env.NEXT_PUBLIC_API_BASE,
+});
```
참고로, axios를 사용하다가 다른 라이브러리를 사용하게 되더라도 axios와 직접적 관계가 없는 부분들이 영향을 받지 않게 하기 위해서 interface 같은 추상 계층을 끼워 넣는답니다.
